stubs/openpyxl/openpyxl/workbook/workbook.pyi

```diff
@@ -28,49 +30,56 @@ class Workbook:
     rels: Incomplete
     calculation: Incomplete
     views: Incomplete
+    # Private, but useful as a reference of what "sheets" can be for other types
+    _sheets: list[_WorkbookChild | ReadOnlyWorksheet]
     def __init__(self, write_only: bool = ..., iso_dates: bool = ...) -> None: ...
     @property
-    def epoch(self) -> datetime: ...
+    def epoch(self) -> datetime.datetime: ...
     @epoch.setter
-    def epoch(self, value: datetime) -> None: ...
+    def epoch(self, value: datetime.datetime) -> None: ...
     @property
     def read_only(self) -> bool: ...
     @property
     def data_only(self) -> bool: ...
     @property
     def write_only(self) -> bool: ...
     @property
-    def excel_base_date(self) -> datetime: ...
+    def excel_base_date(self) -> datetime.datetime: ...
     @property
-    def active(self) -> _WorkbookChild | None: ...
+    def active(self) -> _WorkbookChild: ...
```

Comment:
The returned object will either be a Worksheet (likely case), WriteOnlyWorksheet or Chartsheet. These are all children of _WorkbookChild, but has many more methods associated.

Is is possible to be more specific in the return type, so that method/property code completion can be more accurate?

Or is it expected that the coder would need to specify the type they expect?

Reply:
The thing about union return types, is that you're stuck doing `isinstance` checks anyway. (see https://github.com/python/mypy/issues/1693 and https://github.com/python/typing/issues/566 ).

But if the known types are limited, I can narrow it down further. If you're only interested in the shared properties/methods. It won't change much.

If it wasn't for the fact that you can use an index to set the active sheet, you could restrict it to `Worksheet` or `Chartsheet` (not `WriteOnlyWorksheet` because it doesn't seem to have a `sheet_state).` But with an index you can set any worksheet as the active one.